### DISCUSSION TITLE
Issue with Azure AD App Registration

### DISCUSSION THREAD

**Discussion by jonnynic** [netbox-community/netbox#10063] [Q&A]
Hi Guys,

I would appreciate some help with Netbox Azure AD Authentication.

I keep getting the following error:

![image](https://user-images.githubusercontent.com/111534600/185479068-037ba873-5dea-4510-b76a-3bf4a6ce11d9.png)

Here is my current config from configuration.py

REMOTE_AUTH_ENABLED = True
REMOTE_AUTH_BACKEND = 'social_core.backends.azuread.AzureADOAuth2'
SOCIAL_AUTH_REDIRECT_IS_HTTPS = True
REMOTE_AUTH_AUTO_CREATE_USER = True
SOCIAL_AUTH_AZUREAD_OAUTH2_KEY = '###'
SOCIAL_AUTH_AZUREAD_OAUTH2_SECRET = '###'

I have obfuscated the KEY and SECRET intentionally but have confirmed that these are correct and have viewed all the previous discussions regarding this issue and verified that everything is correct.

Any help would be greatly appreciated!

**Comment by bogdancordos** [comment]
Hi,
I would suggest upgrading your NetBox as this looks like a Django error.

**Reply by candlerb** [reply]
Netbox 3.2.7 is almost the latest version available; to me it looks like a python-social-auth problem. 

You'll need to find out if anyone has managed to get Netbox authenticating against Azure AD. The fact that this backend is for OAuth2 (for resource authorization) rather than OpenID Connect might be an issue.

There are also various different options which you could try, e.g. linking to a specific tenant.
https://python-social-auth.readthedocs.io/en/latest/backends/azuread.html

**Comment by bogdancordos** [comment]
`social_core.backends.azuread.AzureADOAuth2` works fine for me.
I have no extra config compared to @jonnynic 
running:
`"python-version": "3.8.10"`
`"social_django": "5.0.0"`

**Comment by shauntilyard** [comment]
I set this up today and hit the same issue you have. I had to go in an manually authorise the app in Azure AD to get it working. Look under the "Enterprise App" section in Azure and find you App and then "Grant Admin consent......" under permissions.
